src/shipit_code_coverage/tests/test_report_generators.py

```diff
@@ -1,17 +1,74 @@
 # -*- coding: utf-8 -*-
 import json
 import os
+import shutil
+import time
+from datetime import datetime
+
+import hglib
 
 from shipit_code_coverage import report_generators
 
 
+def create_fake_repo(tmp):
+
+    def tobytes(x):
+        return bytes(x, 'ascii')
+
+    dest = os.path.join(tmp, 'repos')
+    local = os.path.join(dest, 'local')
+    remote = os.path.join(dest, 'remote')
+    for d in [local, remote]:
+        os.makedirs(d)
+        hglib.init(d)
+
+    oldcwd = os.getcwd()
+    os.chdir(local)
+    hg = hglib.open(local)
+
+    files = [{'name': 'mozglue/build/dummy.cpp',
+              'size': 1},
+             {'name': 'toolkit/components/osfile/osfile.jsm',
+              'size': 2},
+             {'name': 'js/src/jit/JIT.cpp',
+              'size': 3},
+             {'name': 'toolkit/components/osfile/osfile-win.jsm',
+              'size': 4}]
+
+    for c in '?!':
+        for f in files:
+            fname = f['name']
+            parent = os.path.dirname(fname)
+            if not os.path.exists(parent):
+                os.makedirs(parent)
+            with open(fname, 'w') as Out:
+                Out.write(c * f['size'])
+            hg.add(files=[tobytes(fname)])
+            hg.commit(message='Commit file {} with {} inside'.format(fname, c))
```

Comment:
Maybe you can force an author here, so you don't need to put it in mercurial.nix?